### ISSUE TITLE
Using custom API url breaks plugin

### ISSUE THREAD

**Issue by RokasMuningis** [WordPress/gutenberg#1531]
So I'm using this code:

```
add_filter( 'rest_url_prefix', function ( $slug ) { 
  return 'api';
});
```

but Gutenberg tries to send request to wp-json, and of course it fails.

**Comment by swissspidy** [comment]
When you open the editor and search for `wpApiSettings` in the source code, what do you see? `wpApiSettings` should be set with the correct prefix automatically by the REST API and that's what the editor uses.

**Comment by RokasMuningis** [comment]
Can't do that, editor doesn't even load :/

**Comment by swissspidy** [comment]
Doesn't matter, you just have to open the page and view the page source (i.e. using developer tools, Cmd+alt+u, right click -> view source, etc.).

Also, in the developer tools (usually opens by pressing F12 or cmd+alt+i) you can access the browser console to see what error messages you have in there.

**Comment by RokasMuningis** [comment]
Oh, by editor you meant DevTools? 

wpApiSettings is equal to 
```
{"root":"http:\/\/api.example.com\/wp-json\/","nonce":"f2ce2648e7","versionString":"wp\/v2\/"};
```

**Comment by swissspidy** [comment]
Hmm, strange.

I just added your code

```
add_filter( 'rest_url_prefix', function ( $slug ) { 
  return 'api';
});
```

to the functions.php file of my active theme.

`wpApiSettings` contains `{"root":"http:\/\/api.example.com\/api\/" …`

In the editor, I got a warning because the URL couldn't be found. After that, I saved my permalink settings to properly update the rewrite rules.

Everything works now as expected and I can use the editor without any problems.
